Source/ARTRealtime.m

```diff
@@ -810,12 +811,28 @@ - (BOOL)isActive {
 - (void)sendImpl:(ARTProtocolMessage *)msg callback:(void (^)(ARTStatus *))cb {
     if (msg.ackRequired) {
         msg.msgSerial = [NSNumber numberWithLongLong:self.msgSerial++];
```

Comment:
Ah right, I didn't notice we need the updated value. I think `msg.msgSerial = [NSNumber numberWithLongLong:self.msgSerial + 1]` here and then below just `self.msgSerial++;` would be the simplest.